server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -861,6 +862,7 @@ protected void abandonSegment(final long truncatedTime, final Sink sink)
         );
         for (FireHydrant hydrant : sink) {
           cache.close(SinkQuerySegmentWalker.makeHydrantCacheIdentifier(hydrant));
+          hydrant.getSegment().close();
```

Comment:
IMO, another PR is ok. Or maybe even skipping it altogether, and instead, migrating users of Plumbers to Appenderator (which is meant to be an improved replacement).

The flushing plumber isn't really meant to be used in production anyway (I think it's not even documented). It was meant to be a way to set up some realtime demos that just throw away data after a period of time.